pkg/test/masteruserrecord/master_user_record.go

```diff
@@ -16,46 +19,106 @@ import (
 type MurModifier func(mur *toolchainv1alpha1.MasterUserRecord)
 type UaInMurModifier func(targetCluster string, mur *toolchainv1alpha1.MasterUserRecord)
 
+// DefaultNSTemplateTier the default NSTemplateTier used to initialize the MasterUserRecord
+var DefaultNSTemplateTier = toolchainv1alpha1.NSTemplateTier{
+	ObjectMeta: metav1.ObjectMeta{
+		Namespace: test.HostOperatorNs,
+		Name:      "basic",
+	},
+	Spec: toolchainv1alpha1.NSTemplateTierSpec{
+		Namespaces: []toolchainv1alpha1.NSTemplateTierNamespace{
+			{
+				TemplateRef: "basic-dev-123abc",
+			},
+			{
+				TemplateRef: "basic-code-123abc",
+			},
+			{
+				TemplateRef: "basic-stage-123abc",
+			},
+		},
+		ClusterResources: &toolchainv1alpha1.NSTemplateTierClusterResources{
+			TemplateRef: "basic-clusterresources-654321a",
+		},
+	},
+}
+
+// DefaultNSTemplateSet the default NSTemplateSet used to initialize the MasterUserRecord
+var DefaultNSTemplateSet = toolchainv1alpha1.NSTemplateSet{
+	ObjectMeta: metav1.ObjectMeta{
+		Namespace: test.HostOperatorNs,
+		Name:      DefaultNSTemplateTier.Name,
+	},
+	Spec: toolchainv1alpha1.NSTemplateSetSpec{
+		TierName: DefaultNSTemplateTier.Name,
+		Namespaces: []toolchainv1alpha1.NSTemplateSetNamespace{
+			{
+				TemplateRef: "basic-dev-123abc",
+			},
+			{
+				TemplateRef: "basic-code-123abc",
+			},
+			{
+				TemplateRef: "basic-stage-123abc",
+			},
+		},
+		ClusterResources: &toolchainv1alpha1.NSTemplateSetClusterResources{
+			TemplateRef: "basic-clusterresources-654321a",
+		},
+	},
+}
+
 func NewMasterUserRecord(userName string, modifiers ...MurModifier) *toolchainv1alpha1.MasterUserRecord {
-	userId := uuid.NewV4().String()
+	userID := uuid.NewV4().String()
+	hash, _ := computeTemplateRefsHash(DefaultNSTemplateTier) // we can assume the JSON marshalling will always work
 	mur := &toolchainv1alpha1.MasterUserRecord{
 		ObjectMeta: metav1.ObjectMeta{
-			Name:      userName,
 			Namespace: test.HostOperatorNs,
+			Name:      userName,
+			Labels: map[string]string{
+				templateTierNameLabel(test.MemberClusterName): DefaultNSTemplateTier.Name,
+				templateTierHashLabel(test.MemberClusterName): hash,
+			},
 		},
 		Spec: toolchainv1alpha1.MasterUserRecordSpec{
-			UserID:       userId,
+			UserID:       userID,
 			UserAccounts: []toolchainv1alpha1.UserAccountEmbedded{newEmbeddedUa(test.MemberClusterName)},
 		},
 	}
 	Modify(mur, modifiers...)
 	return mur
 }
 
+// templateTierNameLabel returns the label key to specify the tier templates in use on the given cluster
+func templateTierNameLabel(cluster string) string {
+	return toolchainv1alpha1.LabelKeyPrefix + cluster + "-templates-tier"
```

Comment:
The same question as in https://github.com/codeready-toolchain/host-operator/pull/221#discussion_r439080801
Do we really have to use cluster names in these labels?

Reply:
I responded in https://github.com/codeready-toolchain/host-operator/pull/221#discussion_r440610354 and I'll apply the necessary changes here as well ;)